tests/simulation/test_circuit.py

```diff
@@ -95,3 +106,18 @@ def test_alternative_apply_cnot_three_qubits(BackendClass):
     prop_exp = np.array([0, 0, 0, 0, 0, 0, 0, 1**2])
     assert all(abs(abs(prop_found[i]) - abs(prop_exp[i])) < 0.2 for i in range(8)), \
         f"Propagation failed. Expected prop {prop_exp} vs. found {prop_found}."
+    
+def test_non_close_gate_binary_circuit():
```

Comment:
@paolodr98 Please remove this function and the whitespaces in case the function is not used.

Reply:
Pro tip: You can use an IDE like Intellj to find errors in the code like spelling errors, unused code, unreachable code, unassigned variables etc.

Reply:
You can probably get a student license for free for another year.